spanner/transaction.go

```diff
@@ -133,13 +133,13 @@ func (t *txReadOnly) ReadWithOptions(ctx context.Context, table string, keys Key
 			limit = opts.Limit
 		}
 	}
-	return stream(
+	return streamWithReplaceSessionFunc(
 		contextWithOutgoingMetadata(ctx, sh.getMetadata()),
 		sh.session.logger,
 		func(ctx context.Context, resumeToken []byte) (streamingReceiver, error) {
 			return client.StreamingRead(ctx,
 				&sppb.ReadRequest{
-					Session:     sid,
+					Session:     t.sh.getID(),
```

Comment:
Should we delete the unused `sid` code above?

Reply:
Shouldn't `t.sh.getID()` and `sid` be same here? So why do we use `t.sh.getID()`?

Reply:
Yep, good point. Removed it now.